File: 283. Move Zeroes/py/moveZeroes.py
'''
LeetCode problem 283. Move Zeroes
https://leetcode.com/problems/move-zeroes/description
'''
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        Time complexity: O(n)
        Space complexity: O(1)
        """
        nums_count = len(nums)
        insert_index = 0
        for n in range(nums_count):
            if nums[n] != 0:
                nums[insert_index] = nums[n]
                insert_index += 1
        for z in range(insert_index, nums_count):
            nums[z] = 0

File: 283. Move Zeroes/cs/MoveZeroes/MoveZeroes.cs
/*
LeetCode problem 283. Move Zeroes
https://leetcode.com/problems/move-zeroes/description
*/

namespace LeetCode.MoveZeroes;

public class Solution
{
    public void MoveZeroes(int[] nums)
    {
        /*
        Do not return anything, modify nums in-place instead.
        Time complexity: O(n)
        Space complexity: O(1)
        */
        int numsCount = nums.Length;
        int insertIndex = 0;
        for (int n = 0; n < numsCount; n++)
        {
            if (nums[n] != 0)
            {
                nums[insertIndex] = nums[n];
                insertIndex++;
            }
        }
        for (int z = insertIndex; z < numsCount; z++)
        {
            nums[z] = 0;
        }
    }
}